pygmt/helpers/utils.py

```diff
@@ -848,7 +873,17 @@ def sequence_join(
                 f"but got {len(value)} values."
             )
             raise GMTInvalidInput(msg)
-        return sep.join(str(v) for v in value)
+        # 'str(v)' produces a string like '2024-01-01 00:00:00' for some datetime-like
+        # objects (e.g., datetime.datetime, pandas.Timestamp), which contains a space.
+        # If so, use np.datetime_as_string to convert it to ISO 8601 string format
+        # YYYY-MM-DDThh:mm:ss.ffffff.
+        _value = [
+            np.datetime_as_string(np.asarray(item, dtype=np.datetime64))
+            if " " in str(item)
+            else item
+            for item in value
+        ]
+        return sep.join(str(v) for v in _value)
```

Comment:
We have some similar code here from #2885/#562:

https://github.com/GenericMappingTools/pygmt/blob/bb6e084bc946beb6870e300097aa8d03c6c7ee8d/pygmt/helpers/decorators.py#L722-L732

should there be a common function maybe to avoid duplication?

Reply:
Yes, the new codes are modified from the one you posted. I prefer to keep the `kwargs_to_strings` decorator unchanged, since we're working towards removing this decorator.